uint64 time
uint8 color
float64 x_coord
float64 y_coord
float64 angle